src/main/java/net/pms/dlna/DLNAResource.java

```diff
@@ -2892,13 +2916,26 @@ private void addImageResource(StringBuilder sb, DLNAImageResElement resElement)
 		if (!resElement.isResolutionKnown() && DLNAImageProfile.JPEG_RES_H_V.equals(resElement.getProfile())) {
 			throw new IllegalArgumentException("Resolution cannot be unknown for DLNAImageProfile.JPEG_RES_H_V");
 		}
-		String url = DLNAImageProfile.JPEG_RES_H_V.equals(resElement.getProfile()) ?
-			"JPEG_RES" + resElement.getWidth() + "x" + resElement.getHeight() :
-			resElement.getProfile().toString();
-		url = resElement.isThumbnail() ? getThumbnailURL(url) : getURL(url + "_");
+		String url;
+		if (resElement.isThumbnail()) {
+			url = getThumbnailURL(resElement.getProfile());
+		} else {
+			url = getURL(
+				(DLNAImageProfile.JPEG_RES_H_V.equals(resElement.getProfile()) ?
+					"JPEG_RES" + resElement.getWidth() + "x" + resElement.getHeight() :
+					resElement.getProfile().toString()
+				) + "_"
+			);
+		}
 		if (StringUtils.isNotBlank(url)) {
 			String ciFlag;
-			if (resElement.getCiFlag() == null) {
+			/*
+			 * Some Panasonic TV's can't handle if the thumbnails have the CI
+			 * flag set to 0 while the main resource doesn't have a CI flag.
+			 * DLNA dictates that a missing CI flag should be interpreted as
+			 * if it were 0, so the result should be the same.
+			 */
+			if (resElement.getCiFlag() == null || resElement.getCiFlag() == 0) {
```

Comment:
@valib I wasn't working before without any complaints. First of all you had to had a renderer setting to decide if the thumbnail should be PNG or JPEG, second it still didn't work very well on all renderers. 

#1061 changed all this, in that it offers multiple thumbnail formats for any given resource (both JPEG, PNG and sometimes GIF, in multiple resolutions). That's so that all renderers should find a thumbnail they can understand, and also because this is how DLNA is supposed to work.

The CI flag is irrelevant as long as you only have one entry/URL for one resource, because the renderer has no choice and must pick that one. If it doesn't understand/support that one it will fail. With #1061 and multiple choices, the renderer is, according to DLNA, supposed to pick a thumbnail format profile it can use rom multiple, and then the CI flag has a purpose in that it tells the renderer which of the thumbnails that's transcoded/converted. The idea is that if the renderer supports one of the CI = 0 entries, it should pick that both because the server doesn't have to convert it and because the quality is better without conversion. But, if it can't understand the format profile with CI = 0, it will pick one of the CI = 1 profiles. 

By removing the CI flag the renderers can't make an "informed" decision. But, since the absence of the CI flag means the same as CI = 0 (according to DLNA), and your renderer doesn't pick the thumbnail by mistake when done this way, that's the best solution. More intelligent renderers still have the CI flag available for thumbnails where CI = 1, so it shouldn't make a difference if they follow the standard. I'm sure some don't, but there's nothing much we can do about that.